python/ray/job_config.py

```diff
@@ -1,81 +1,138 @@
-        py_driver_sys_path: A list of directories that
-            specify the search path for python workers.
+            See :ref:`Ray cross-language programming <cross_language>` for more details.
+        runtime_env: A :ref:`runtime environment <runtime-environment>` dictionary.
+        metadata: An opaque metadata dictionary.
+        ray_namespace: A :ref:`namespace <namespaces-guide>`
+            is a logical grouping of jobs and named actors.
+        default_actor_lifetime: The default value of actor lifetime,
+            can be "detached" or "non_detached".
+
+    Examples:
+
+        .. testcode::
+
+            import ray
+            from ray.job_config import JobConfig
+
+            ray.init(job_config=JobConfig(default_actor_lifetime="non_detached"))
     """
 
     def __init__(
         self,
-        jvm_options: List[str] = None,
-        code_search_path: List[str] = None,
-        runtime_env: dict = None,
-        client_job: bool = False,
+        jvm_options: Optional[List[str]] = None,
+        code_search_path: Optional[List[str]] = None,
+        runtime_env: Optional[dict] = None,
         metadata: Optional[dict] = None,
         ray_namespace: Optional[str] = None,
         default_actor_lifetime: str = "non_detached",
-        py_driver_sys_path: List[str] = None,
+        _client_job: bool = False,
+        _py_driver_sys_path: List[str] = None,
     ):
+        #: The jvm options for java workers of the job.
         self.jvm_options = jvm_options or []
+        #: A list of directories or jar files that
+        #: specify the search path for user code.
         self.code_search_path = code_search_path or []
         # It's difficult to find the error that caused by the
         # code_search_path is a string. So we assert here.
         assert isinstance(self.code_search_path, (list, tuple)), (
             f"The type of code search path is incorrect: " f"{type(code_search_path)}"
         )
-        self.client_job = client_job
+        self._client_job = _client_job
+        #: An opaque metadata dictionary.
         self.metadata = metadata or {}
+        #: A namespace is a logical grouping of jobs and named actors.
         self.ray_namespace = ray_namespace
         self.set_runtime_env(runtime_env)
         self.set_default_actor_lifetime(default_actor_lifetime)
-        self.py_driver_sys_path = py_driver_sys_path or []
+        # A list of directories that specify the search path for python workers.
+        self._py_driver_sys_path = []
 
     def set_metadata(self, key: str, value: str) -> None:
+        """Add key-value pair to the metadata dictionary.
+
+        If the key already exists, the value will be overwritten to the new value.
+
+        Examples:
+
+            .. testcode::
+
+                import ray
+                from ray.job_config import JobConfig
+
+                job_config = JobConfig()
+                job_config.set_metadata("key1", "value1")
```

Comment:
Nit: are there more practical values we could use other than "key1" and "value1"? I think it'd make this example more instructive